mmv1/products/containerattached/terraform.yaml

```diff
@@ -0,0 +1,64 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Provider::Terraform::Config
+overrides: !ruby/object:Overrides::ResourceOverrides
+  Cluster: !ruby/object:Overrides::Terraform::ResourceOverride
+    # Don't generate the LRO handling code because it doesn't account for regional services.
+    autogen_async: false
+    id_format: "projects/{{project}}/locations/{{location}}/attachedClusters/{{name}}"
+    import_format: ["projects/{{project}}/locations/{{location}}/attachedClusters/{{name}}"]
+    examples:
+      - !ruby/object:Provider::Terraform::Examples
+        name: "container_attached_cluster_minimal"
+        primary_resource_name: "fmt.Sprintf(\"basic%s\", context[\"random_suffix\"])"
+        primary_resource_id: "primary"
+        vars:
+          name: "basic"
+      - !ruby/object:Provider::Terraform::Examples
+        name: "container_attached_cluster_full"
+        primary_resource_name: "fmt.Sprintf(\"basic%s\", context[\"random_suffix\"])"
+        primary_resource_id: "primary"
+        vars:
+          name: "basic"
+    properties:
+      # The proto format of the authorization field is cumbersome to express in TF, so use a custom expander
+      # and flattener to massage the data.
+      authorization: !ruby/object:Overrides::Terraform::PropertyOverride
+        custom_expand: 'templates/terraform/custom_expand/containerattached_cluster_authorization_user.go.erb'
+        custom_flatten: 'templates/terraform/custom_flatten/containerattached_cluster_authorization_user.go.erb'
+      # Convenience expander to prepend "projects/" to fleet project numbers if not already present.
+      # Custom diff to take this convenience into account.
+      fleet.project: !ruby/object:Overrides::Terraform::PropertyOverride
+        diff_suppress_func: suppressAttachedClustersFleetProjectDiff
+        custom_expand: 'templates/terraform/custom_expand/containerattached_cluster_fleet_project.go.erb'
+      # If the user doesn't specify a loggingConfig, the server will supply a default value.
+      # No diff suppression because it is still possible to explicitly clear the config by passing
+      # an empty value.
+      loggingConfig: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
+      # If the user omits the monitoring config, the server will still respond with an empty config, which
+      # causes diff problems.
+      monitoringConfig: !ruby/object:Overrides::Terraform::PropertyOverride
+        diff_suppress_func: 'suppressAttachedClustersMonitoringConfigDiff'
```

Comment:
It looks like this is similar in purpose to [`emptyOrUnsetBlockDiffSuppress`](https://github.com/hashicorp/terraform-provider-google/blob/6be5859718f35b1af72d4ca00a610e488ea3d854/google/common_diff_suppress.go#L113), which has a note:

> Using Optional + Computed is *strongly* preferred to this DSF, as it's more well understood and resilient to API changes.

Given that - default_from_api may be better here.

Reply:
Yeah, `default_from_api` seems to work and is less confusing.